sklearn/linear_model/tests/test_ridge.py

```diff
@@ -314,37 +338,40 @@ def test_ridge_regression_unpenalized(
         tol=1e-15 if solver in ("sag", "saga") else 1e-10,
         random_state=global_random_seed,
     )
-
     model = Ridge(**params)
-    # Note that cholesky might give a warning: "Singular matrix in solving dual
-    # problem. Using least-squares solution instead."
+
     if fit_intercept:
         X = X[:, :-1]  # remove intercept
         intercept = coef[-1]
         coef = coef[:-1]
     else:
+        # Consider the centered problem without intercept instead.
+        X = X[:, :-1]
+        coef = coef[:-1]
+        X = X - X.mean(axis=0)
+        y = y - y.mean()
         intercept = 0
-    model.fit(X, y)
 
-    # FIXME: `assert_allclose(model.coef_, coef)` should work for all cases but fails
-    # for the wide/fat case with n_features > n_samples. The current Ridge solvers do
-    # NOT return the minimum norm solution with fit_intercept=True.
-    if n_samples > n_features or not fit_intercept:
-        assert model.intercept_ == pytest.approx(intercept)
-        assert_allclose(model.coef_, coef)
+    # Note that cholesky might give a warning: "Singular matrix in solving dual
+    # problem. Using least-squares solution instead."
```

Comment:
Actually, the warning type and message is not always the same, depending on the shape of the data.

```suggestion
```